components/sup/src/manager/self_updater.rs

```diff
@@ -134,7 +139,7 @@ impl SelfUpdater {
     }
 
     pub fn updated(&mut self) -> Option<PackageInstall> {
-        if let Some(ref mut rx) = self.rx {
+        if let UpdateStrategy::Auto(ref mut rx) = self.strategy {
```

Comment:
I'd go with a `match` statement here, instead of the `if let`... it helps carry the intention behind the separate enum variants down into the code here.